Apps/TestProtocol/testBNProtocol.js

```diff
@@ -0,0 +1,130 @@
+const fs = require('fs');
+const shelljs = require('shelljs');
+
+let compatibility = [];
+function execute(command, workingDirectory, callback){
+    const res = shelljs.exec(command, { fatal: false, cwd: workingDirectory, async: false });
+    callback(res.code, res.stdout, res.stderr);
+};
+
+function copy(source, destination) {
+    fs.copyFileSync(source, destination);
+    console.log(`${source} was copied to ${destination}`);
+}
+
+const protocolChecker = "//BRNCHECK\r\nconsole.log(' * Checking Babylon.js / BabylonNative protocols');\r\n"+
+    "if (_native.Engine.PROTOCOL_VERSION !== BABYLON.NativeEngine.PROTOCOL_VERSION) {\r\n"+
+    "console.log(` * Protocol version mismatch: ${_native.Engine.PROTOCOL_VERSION} (Native) !== ${BABYLON.NativeEngine.PROTOCOL_VERSION} (JS)`);\r\n"+
+    "TestUtils.exit(-1); } else { console.log(' * Protocol version OK! '); }\r\n"+
+    "console.error = function(errorMsg) { console.log(errorMsg); TestUtils.exit(-1);}\r\n";
+
+const exeFolder = './BabylonNative/build/Apps/ValidationTests/Release';
+const filesToCopy = [
+    {source:'babylonjs', files:['babylon.max.js', 'babylon.max.js.map']},
+    {source:'babylonjs-gui', files:['babylon.gui.js', 'babylon.gui.js.map']},
+    {source:'babylonjs-loaders', files:['babylonjs.loaders.js', 'babylonjs.loaders.js.map']},
+    {source:'babylonjs-materials', files:['babylonjs.materials.js', 'babylonjs.materials.js.map']}];
+
+const BRNVersions = [
+    {tag:'1.4.0', hash:'5859ffa'},
+    {tag:'1.4.1', hash:'70bb77a'},
+    {tag:'1.4.2', hash:'5990087'},
+    {tag:'1.4.3', hash:'301ab90'},
+    {tag:'1.4.4', hash:'75954f4'},
+    {tag:'1.5.0', hash:'75954f4'},
+    {tag:'1.5.1', hash:'a2cf1c7'},
+    {tag:'1.5.4-preview', hash:'409bc4b'}];
+
+function patchTestScript() {
+    const filePath = `${exeFolder}/Scripts/validation_native.js`;
+    console.log("Patching test script.");
+    const testScript = fs.readFileSync(filePath);
+    if (testScript.includes("//BRNCHECK")) {
+        // already patched
+        return;
+    }
+    const newContent = protocolChecker + testScript;
+    fs.writeFileSync(filePath, newContent);
+}
+
+function checkoutAndBuildBN(tag, hash, callback) {
+    console.log("Git clone.");
+    execute('git clone https://github.com/BabylonJS/BabylonNative.git', './', (error, stdout, stderr) => {
```

Comment:
Oy. Can we use Promises and async/await in this code? It's just a node.js script, right?

Reply:
Yes, it's a Node.js script.